Util/openl-maven-plugin/src/org/openl/rules/maven/TestMojo.java

```diff
@@ -213,16 +218,24 @@ private Summary executeAllAtOnce(String testSourcePath,
                 .setExternalParameters(externalParameters)
                 .build();
 
+            validateOpenApi(factory);
+
             CompiledOpenClass openLRules = factory.getCompiledOpenClass();
             return executeTests(openLRules);
         } finally {
             OpenClassUtil.releaseClassLoader(classLoader);
         }
     }
 
+    private void validateOpenApi(SimpleProjectEngineFactory<?> factory) throws ProjectResolvingException, RulesInstantiationException {
+        OpenApiProjectValidator openApiProjectValidator = new OpenApiProjectValidator();
+        openApiProjectValidator.validate(factory.getProjectDescriptor(), factory.getRulesInstantiationStrategy());
+    }
+
     private Summary executeModuleByModule(String testSourcePath,
             String mainSourcePath,
             boolean hasDependencies) throws MalformedURLException, ProjectResolvingException {
+        log.warn(singleModuleMode ? "singleModuleMode " : "maxModulesInMemory" + "variable was used. OpenApi validation was skipped.");

```

Comment:
this method works in very rare cases - for big projects, to avoid memory issues. In this case we report that open api will not be performed because of insufficient memory